torchci/lib/fetchS3Links.ts

```diff
@@ -0,0 +1,64 @@
+import { Artifact } from "./types";
+import parseXml from "./parseXml";
+
+export default async function fetchS3Links(
+  suiteId: string
+): Promise<Artifact[]> {
+  let result = (await s3(
+    `pytorch/pytorch/${suiteId}/`,
+    "gha-artifacts"
+  )) as any;
+  let prefixes = extractItem(result.ListBucketResult, "CommonPrefixes");
+  const artifacts = [];
+  // If anything was found, go through the results and add the items to
+  // the 'run' object in place
+  if (prefixes && prefixes.length > 0) {
+    for (const prefixItem of prefixes) {
+      let prefix = prefixItem.Prefix["#text"];
+      let result = (await s3(prefix, "gha-artifacts")) as any;
+      let contents = extractItem(result.ListBucketResult, "Contents");
+      for (const content of contents) {
+        let prefix = content.Key["#text"];
+        let url = `https://gha-artifacts.s3.amazonaws.com/${prefix}`;
+        artifacts.push({
+          kind: "s3",
+          name: prefix.split("/").slice(-1),
+          sizeInBytes: parseInt(content.Size["#text"]),
+          expired: false,
+          url,
+        });
+      }
+    }
+  }
+  return artifacts;
+}
+
+async function s3(prefix: string, bucket: string) {
+  // List the gha-artifacts S3 bucket by a specific prefix
+  const url =
```

Comment:
should we just use the s3 js client instead of doing this directly? I think that way we probably can avoid implementing XML parsing logic.

Reply:
Good point! Didn't know this was a sdk. Resolved.